app/src/test/java/tech/relaycorp/gateway/pdc/local/routes/ParcelCollectionRouteTest.kt

```diff
@@ -0,0 +1,114 @@
+import kotlinx.coroutines.flow.emptyFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import tech.relaycorp.gateway.data.model.MessageAddress
+import tech.relaycorp.gateway.domain.endpoint.CollectParcels
+import tech.relaycorp.gateway.pdc.local.utils.ParcelCollectionHandshake
+import tech.relaycorp.gateway.test.WaitAssertions.suspendWaitFor
+import tech.relaycorp.relaynet.messages.control.ParcelDelivery
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import javax.inject.Provider
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class ParcelCollectionRouteTest {
+
+    private val parcelCollectionHandshake = mock<ParcelCollectionHandshake>()
+    private val collectParcels = mock<CollectParcels>()
+    private val route =
+        ParcelCollectionRoute(parcelCollectionHandshake, Provider { collectParcels })
+
+    private val endpointAddress = MessageAddress.of("1234")
+    private val certificate = mock<Certificate>()
+        .also { whenever(it.subjectPrivateAddress).thenReturn(endpointAddress.value) }
+
+    @BeforeEach
+    internal fun setUp() = runBlockingTest {
+        whenever(parcelCollectionHandshake.handshake(any())).thenReturn(listOf(certificate))
+        whenever(collectParcels.getNewParcelsForEndpoints(any())).thenReturn(emptyFlow())
+        whenever(collectParcels.noParcelsToDeliverOrAck).thenReturn(flowOf(false))
+    }
+
+    @Test
+    fun `Server should close if there is not Keep-Alive and no parcel to receive`() =
+        runBlockingTest {
+            whenever(collectParcels.noParcelsToDeliverOrAck).thenReturn(flowOf(true))
+
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(ParcelCollectionRoute.URL_PATH) { incoming, _ ->
+                    val closingFrameRaw = incoming.receive()
+                    assertEquals(FrameType.CLOSE, closingFrameRaw.frameType)
+
+                    val closingFrame = closingFrameRaw as Frame.Close
+                    assertEquals(
+                        CloseReason.Codes.NORMAL,
+                        closingFrame.readReason()!!.knownReason
+                    )
+                    assertEquals(
+                        "All available parcels delivered",
+                        closingFrame.readReason()!!.message
+                    )
+                }
+            }
+        }
+
+    @Test
+    fun `Server should keep connection if Keep-Alive is on`() =
+        runBlockingTest {
+            whenever(collectParcels.noParcelsToDeliverOrAck).thenReturn(flowOf(true))
+
+            testPDCServerRoute(route) {
+                handleWebSocketConversation(
+                    ParcelCollectionRoute.URL_PATH,
+                    { addHeader(ParcelCollectionRoute.HEADER_KEEP_ALIVE, "on") }
+                ) { incoming, _ ->
+                    assertTrue(incoming.isEmpty)
```

Comment:
How is this checking that the connection is kept alive?

This is a tough condition to check I guess. Maybe we need a timeout?